Identity.API/AuthenticationService.cs

```diff
@@ -1,6 +1,111 @@
-﻿namespace Identity.API
+﻿using Identity.BusinessLogic;
+using Identity.Core.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace Identity.API
 {
-    public class AuthenticationService
+    public class AuthenticationService(UserManager<User> userManager, RoleManager<IdentityRole> roleManager, IConfiguration configuration, ILogger<AuthenticationService> logger) : IAuthenticationService
     {
+        public readonly UserManager<User> _userManager = userManager;
+        public readonly RoleManager<IdentityRole> _roleManager = roleManager;
+        public readonly IConfiguration _configuration = configuration;
+
+        public readonly ILogger<AuthenticationService> _logger = logger;
+
+        public async Task<string?> AuthenticateAsync(Login loginCredentials)
+        {
+            _logger.LogInformation("Searching for user with email {email} ", loginCredentials.Email);
+            var user = await _userManager.FindByEmailAsync(loginCredentials.Email);
+
+
+            if (user is null)
+            {
+                _logger.LogDebug("No User");
+                return null;
```

Comment:
Instead of returning null, exceptions should be thrown. And make a global exception handler